packages/truffle-db/src/loaders/artifacts/test/index.ts

```diff
@@ -149,46 +165,150 @@ query getWorkspaceCompilation($id: ID!) {
+      expect(source["contents"]).toEqual(Build[index].source);
+      expect(solcCompilation.contracts[index].name).toEqual(Build[index].contractName);
+    });
+
+    const vyperCompilation =  compilationsQuery[1].data.workspace.compilation
+    expect(vyperCompilation.compiler.version).toEqual(Build[3].compiler.version);
+    expect(vyperCompilation.sources.length).toEqual(1);
+    expect(vyperCompilation.sources[0].id).toEqual(sourceIds[3].id);
+    expect(vyperCompilation.sources[0].contents).toEqual(Build[3].source);
+    expect(vyperCompilation.contracts[0].name).toEqual(Build[3].contractName);
+  });
 
-    const {
-      data: {
-        workspace: {
-          compilation: {
-            compiler: {
-              version
-            },
-            sources,
-            contracts
+  it("loads contract sources", async () => {
+    for(let index in sourceIds) {
+      let {
+        data: {
+          workspace: {
+            source: {
+              contents,
+              sourcePath
+            }
           }
         }
-      }
-    } = await db.query(GetWorkspaceCompilation, { id: expectedId });
-
-    expect(version).toEqual(Build[0].compiler.version);
-    expect(sources.length).toEqual(3);
-    for(let source in sources) {
-      expect(sources[source]["id"]).toEqual(sourceIds[source]["id"]);
-      expect(sources[source].contents).toEqual(Build[source].source);
-      expect(contracts[source].name).toEqual(Build[source].contractName);
+      } = await db.query(GetWorkspaceSource, sourceIds[index]);
+
+      expect(contents).toEqual(Build[index].source);
+      expect(sourcePath).toEqual(Build[index].sourcePath);
+    }
+  });
+
+  it("loads bytecodes", async () => {
+    for(let index in bytecodeIds) {
+      let {
+        data: {
+          workspace: {
+            bytecode: {
+              bytes
+            }
+          }
+        }
+      } = await db.query(GetWorkspaceBytecode, bytecodeIds[index]);
+
+      expect(bytes).toEqual(Build[index].bytecode);
+
     }
+  });
+
+  it("loads contracts", async () => {
+    let contractIds = [];
+
+    for(let index in Build) {
+      let expectedId = generateId({
+        name: Build[index].contractName,
+        abi: { json: JSON.stringify(Build[index].abi) },
+        sourceContract: { index: Build[index].compiler.name === "solc" ? +index : 0},
+        compilation: {
+          id: Build[index].compiler.name === "solc" ? expectedSolcCompilationId : expectedVyperCompilationId
+        }
+      });
 
-  })
+      contractIds.push({ id: expectedId });
+      let {
+        data: {
+          workspace: {
+            contract: {
+              id,
+              name,
+              abi: {
+                json
+              },
+              constructor: {
+                createBytecode: {
+                  bytes
+                }
+              },
+              sourceContract: {
+                source: {
+                  contents,
+                  sourcePath
```

Comment:
`sourcePath` isn't being tested for either; might as well remove the unused variable.